### PR TITLE
PostgreSQL S3 Backup Docker Image

### PR DESCRIPTION
**Pull request by ArmaanT**
This PR:
* Adds a new docker image that will backup all the databases within a PostgreSQL cluster to an S3 bucket
* Adds a GitHub action to publish that image

### CODE REVIEW

File: .github/workflows/pg-s3-backup.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      - master
+    paths:
+      - "docker/pg-s3-backup/**"
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Publish pg-s3-backup
+        uses: elgohr/Publish-Docker-Github-Action@2.18
```
Comment (pawalt) [inline .github/workflows/pg-s3-backup.yaml:14]: Why use this over the `docker/` actions?

File: docker/pg-s3-backup/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,29 @@
+FROM debian:buster-slim as build
+
+# Install aws cli
+RUN apt-get update \
+    && apt-get install -y curl unzip \
+    && curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip" \
```
Comment (pawalt) [inline docker/pg-s3-backup/Dockerfile:6]: Is there any way to specify a version on this curl?

File: .github/workflows/pg-s3-backup.yaml (hunk 1 above)
Comment (ArmaanT) [reply .github/workflows/pg-s3-backup.yaml]: Evidently the official docker GH action was in the wasteland known as the second page of Google. Fixed now.

File: docker/pg-s3-backup/Dockerfile (hunk 2 above)
Comment (ArmaanT) [reply docker/pg-s3-backup/Dockerfile]: Done.

**Comment by ArmaanT** [comment]
Yep, the plan is to run it as an icarus cronjob

**Pull request merged**